# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1460611665 -36000
#      Thu Apr 14 15:27:45 2016 +1000
# Node ID 7c0595c722cadafb4047e9025cadbe346c017c9b
# Parent  0822917dfcf52b43cc1f53d744d4e5ab841ab894
Remove duplicate menu actions after replacing <unnamed> edit tab (bug #47692).

* file-editor-tab.cc (file_editor::request_open_file):
  Suppress call to add_file_editor_tab when reusing a tab.

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -452,37 +452,41 @@ file_editor::request_open_file (const QS
               emit fetab_set_focus (tab);
               focus ();
             }
         }
       else
         {
           file_editor_tab *fileEditorTab = 0;
           // Reuse <unnamed> tab if it hasn't yet been modified.
+          bool reusing = false;
           tab = find_tab_widget ("");
           if (tab)
             {
               fileEditorTab = static_cast<file_editor_tab *>(tab);
               if (fileEditorTab->qsci_edit_area ()->isModified ())
                 fileEditorTab = 0;
+              else
+                reusing = true;
             }
 
           // If <unnamed> was absent or modified, create a new tab.
           if (!fileEditorTab)
             fileEditorTab = new file_editor_tab ();
 
           if (fileEditorTab)
             {
               fileEditorTab->set_encoding (encoding);
               QString result = fileEditorTab->load_file (openFileName);
               if (result == "")
                 {
                   // Supply empty title then have the file_editor_tab update
                   // with full or short name.
-                  add_file_editor_tab (fileEditorTab, "");
+                  if (!reusing)
+                    add_file_editor_tab (fileEditorTab, "");
                   fileEditorTab->update_window_title (false);
                   // file already loaded, add file to mru list here
                   QFileInfo file_info = QFileInfo (openFileName);
                   handle_mru_add_file (file_info.canonicalFilePath (),
                                        encoding);
 
                   if (line > 0)
                     {
